core/model/modx/rest/modrestservice.class.php

```diff
@@ -242,8 +242,10 @@ protected function getController() {
      */
     public function iterateDirectories($pattern, $flags = 0) {
         $files = glob($pattern, $flags);
-        foreach (glob(dirname($pattern).'/*', GLOB_ONLYDIR|GLOB_NOSORT) as $dir) {
-            $files = array_merge($files, $this->iterateDirectories($dir.'/'.basename($pattern), $flags));
+        if ($dirs = glob(dirname($pattern).'/*', GLOB_ONLYDIR|GLOB_NOSORT)) {
```

Comment:
It's a matter of personal preference and this practice I learn from MODX sources. For example: [1](https://github.com/modxcms/revolution/blob/53a0086fbdfecedc38e3d83582828ee50044c30b/core/model/modx/modaccessibleobject.class.php#L48), [2](https://github.com/modxcms/revolution/blob/53a0086fbdfecedc38e3d83582828ee50044c30b/core/model/modx/modaccessibleobject.class.php#L81), [3](https://github.com/modxcms/revolution/blob/53a0086fbdfecedc38e3d83582828ee50044c30b/core/model/modx/modaccessibleobject.class.php#L130), [etc](https://github.com/modxcms/revolution/blob/53a0086fbdfecedc38e3d83582828ee50044c30b/core/model/modx/modresource.class.php#L317).

Anyway, you can change my PR as you want.